### ISSUE TITLE
postgrespro-1c-14 и pg_probackup

### ISSUE THREAD

**Issue by bsn5** [postgrespro/pg_probackup#544]
Здравствуйте, скажите, пожалуйста, поддерживает ли pg_probackup-{std,ent}-{13,12,11,10,9.6} сервер postgrespro-1c-14 с сайта 1c.postgres.ru.

Я не могу выполнить команду:
pg_probackup-std-13 add-instance -B /home/pavlovt/pg_probackup_catalog -D /var/lib/pgpro/1c-14/data --instance UNF --remote-host=deb10-12-xfce-cstm-1c --remote-port=22 --remote-user=postgres --remote-path=/usr/bin --ssh-options="-o ServerAliveCountMax=5 -o ServerAliveInterval=60"

Я все время получаю ошибку:

![MqLQgMuk](https://user-images.githubusercontent.com/44580944/193020720-dad43ef9-6f64-4efb-804c-37dd3ba264fb.jpg)

Контрольные суммы пробовал и включать и отключать в кластере, не помогает, подскажите, пожалуйста, в чем может быть проблема?

**Comment by funny-falcon** [comment]
Думаю, нужно всё-таки попробовать pg_probackup-std-14.

**Comment by bsn5** [comment]
В репозитории его нет.

**Comment by bsn5** [comment]
Установил postgrespro-1c-13, включил контрольные суммы в кластере, та же проблема.

**Comment by funny-falcon** [comment]
Мне тут подсказывают, что всё же нужно взять ванильную версию. Т.е. скомпилировать pg_probackup с open source версией PostgreSQL соответствующей версии: https://github.com/postgrespro/pg_probackup#pg_probackup-for-vanilla-postgresql

**Comment by bsn5** [comment]
По ваниле wget не отрабатывает, пишет, что не может прочитать списко источников.

**Comment by bsn5** [comment]
Не правильно сказал, wget отработал нормально sudo apt update не отрабатывает.

**Comment by bsn5** [comment]
Не помогло, к сожалению, vanila выдает все ту же ошибку.

**Comment by bsn5** [comment]
Извините, за ошибку, сработало, спасибо вам большое.

**Comment by bsn5** [comment]
Вы не могли бы еще подсказать. Уже при восстановлении базы 1С из резервной копии выходит ошибка в лог:
![photo_2022-10-03_12-04-35](https://user-images.githubusercontent.com/44580944/193519700-1ec2042f-3f47-43a4-a981-7018e8eaa510.jpg)
Т.е. получается сервер при восстановлении запускается, но тем не менее до конца восстановление не доходит.

**Comment by slothfk** [comment]
> Т.е. получается сервер при восстановлении запускается, но тем не менее до конца восстановление не доходит.

Там же в логе написано что надо сделать, а именно
```
sudo -u postgres psql -c 'select pg_wal_replay_resume()'
```
конкретно в данном случае
Пока эту команду не выполните, кластер "на паузе", т.е. в режиме "только чтение" ...

**Comment by bsn5** [comment]
Да, спасибо, я уже разобрался, но я файле конфигурации поставил значение promote вместо pause.

**Comment by Sergey-Nova** [comment]
И все таки, можно ли просить о скомпилированной версии pg_probackup для 14 версии (ванильная)?  Компилировать на С, для меня будет новизной.

**Comment by kulaginm** [comment]
@Sergey-Nova , в этом запросе обсуждалась работа с 1С версией.

**Comment by Sergey-Nova** [comment]
Я про работу с 1С версией и имел ввиду. Открыть новую тему?

**Comment by kulaginm** [comment]
@Sergey-Nova , нет, работа с 1С -- это эта тема.
Как ранее и говорилось в этой теме, установка pg_probackup описана в https://github.com/postgrespro/pg_probackup#pg_probackup-for-vanilla-postgresql

**Comment by Sergey-Nova** [comment]
Давайте я опишу вопрос. Есть Postgres14 (1C), на Windows. Для установки требуется pg_probackup для 14. Его для скачивания нет - максимальный 13. Ранее пользовался всем для 12, работало (ванильная).
Возможно ли выложить, версию 2.4.15, ванильную, для 14 PG?

**Comment by kulaginm** [comment]
Поддержка 14-й версии postgres появилась только в pg_probackup-2.5.2, поэтому 2.4.15 для 14-й никогда не собиралась (и не будет собираться).
Зачем вам нужна именно 2.4.15? Вы пробовали воспользоваться версией 2.5.8 (https://github.com/postgrespro/pg_probackup/releases/tag/2.5.8)?

**Comment by Sergey-Nova** [comment]
Т.е. отдельно ванильной не надо для 2.5.8?

**Comment by Sergey-Nova** [comment]
И еще, вопрос чайника, возможно ли для pg_probackup сжатие архива "на лету" в среде WinServer?

**Comment by Sergey-Nova** [comment]
Да, не заработал 2.5.8. у меня, жаль.

**Comment by Sergey-Nova** [comment]
Нет, заработал. С сетевой папкой не заработал.
